### PR TITLE
Pagination for transactions

### PR DESCRIPTION
**Pull request by tsehelnyk**

### CODE REVIEW

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/AccountService.java [hunk 1]
```diff
@@ -33,6 +33,10 @@
 @Slf4j
 @RequiredArgsConstructor
 public class AccountService {
+    //TODO Fix hardcoded values
+    private static final int PAGE = 1;
+    private static final int TRANSACTIONS_PER_PAGE = 500;
```
Comment (valb3r) [inline fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/AccountService.java:38]: Passing pageSize as null should force returning all (or provider default value - i.e. 500 for the certain provider) transactions. So this should be null and unnecessary

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/TransactionService.java [hunk 2]
```diff
@@ -31,6 +31,10 @@
 @Slf4j
 @RequiredArgsConstructor
 public class TransactionService {
+    //TODO Fix hardcoded values
+    private static final int PAGE = 1;
+    private static final int TRANSACTIONS_PER_PAGE = 500;
```
Comment (valb3r) [inline fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/TransactionService.java:36]: Passing pageSize as null should force returning all (or provider default value - i.e. 500 for the certain provider) transactions. So this should be null and unnecessary

File: firefly-exporter/src/main/java/de/adorsys/opba/fireflyexporter/service/FireFlyTransactionExporter.java [hunk 3]
```diff
@@ -50,6 +51,11 @@
     private final TransactionCategorizer categorizer;
     private final TransactionExportJobRepository exportJobRepository;
 
+    @Value("${firefly.pages.page:1}")
```
Comment (valb3r) [inline firefly-exporter/src/main/java/de/adorsys/opba/fireflyexporter/service/FireFlyTransactionExporter.java:54]: Same as above

**Review by valb3r — CHANGES_REQUESTED** [review]

File: opba-protocols/xs2a-protocol/src/main/java/de/adorsys/opba/protocol/xs2a/service/xs2a/dto/Xs2aTransactionParameters.java [hunk 4]
```diff
@@ -45,6 +45,18 @@
     @NotNull(message = "{no.ctx.dateTo}")
     private LocalDate dateTo;
 
+    /**
+     * Result page number.
+     */
+    @NotNull(message = "{no.ctx.page}")
+    private Integer page;
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol/src/main/java/de/adorsys/opba/protocol/xs2a/service/xs2a/dto/Xs2aTransactionParameters.java:52]: These ARE nullable

File: opba-protocols/xs2a-protocol/src/main/java/de/adorsys/opba/protocol/xs2a/service/xs2a/dto/Xs2aTransactionParameters.java [hunk 5]
```diff
@@ -45,6 +45,18 @@
...
+
+    /**
+     * Records per page.
+     */
+    @NotNull(message = "{no.ctx.perPage}")
+    private Integer perPage;
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol/src/main/java/de/adorsys/opba/protocol/xs2a/service/xs2a/dto/Xs2aTransactionParameters.java:58]: These ARE nullable

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/AccountService.java [hunk 6]
```diff
@@ -33,6 +33,10 @@
 @Slf4j
 @RequiredArgsConstructor
 public class AccountService {
+    //TODO Fix hardcoded values
+    private static final int PAGE = 1;
```
[the remark this answers is not in the document]
Comment (tsehelnyk) [reply fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/AccountService.java:37]: Deleted unnecessary consts, changed fields to null.

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/TransactionService.java [hunk 7]
```diff
@@ -31,6 +31,10 @@
 @Slf4j
 @RequiredArgsConstructor
 public class TransactionService {
+    //TODO Fix hardcoded values
+    private static final int PAGE = 1;
```
[the remark this answers is not in the document]
Comment (tsehelnyk) [reply fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/TransactionService.java:35]: Deleted unnecessary consts, changed fields to null.

File: firefly-exporter/src/main/java/de/adorsys/opba/fireflyexporter/service/ConsentService.java [hunk 8]
```diff
@@ -30,6 +31,11 @@
     private final RedirectStateRepository redirectStateRepository;
     private final OpenBankingConfig bankingConfig;
 
+    @Value("${firefly.pages.page:1}")
```
[the remark this answers is not in the document]
Comment (tsehelnyk) [reply firefly-exporter/src/main/java/de/adorsys/opba/fireflyexporter/service/ConsentService.java:34]: Deleted unnecessary vars, changed fields to null.

File: firefly-exporter/src/main/java/de/adorsys/opba/fireflyexporter/service/ConsentService.java [hunk 9]
```diff
@@ -30,6 +31,11 @@
...
+    private int page;
+    @Value("${firefly.pages.perPage:500}")
```
[the remark this answers is not in the document]
Comment (tsehelnyk) [reply firefly-exporter/src/main/java/de/adorsys/opba/fireflyexporter/service/ConsentService.java:36]: Deleted unnecessary vars, changed fields to null.

File: opba-facade-protocol-api-shared/src/main/java/de/adorsys/opba/protocol/api/dto/result/body/TransactionsResponseBody.java [hunk 10]
```diff
@@ -25,6 +25,11 @@
    */
   private AnalyticsResult analytics;
 
+  /**
+   * Information for pagination
```
[the remark this answers is not in the document]
Comment (tsehelnyk) [reply opba-facade-protocol-api-shared/src/main/java/de/adorsys/opba/protocol/api/dto/result/body/TransactionsResponseBody.java:29]: fixed

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/AccountService.java (hunk 1 above)
Comment (tsehelnyk) [reply fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/AccountService.java:38]: (duplicated comment)
Deleted unnecessary vars, changed fields to null.

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/TransactionService.java (hunk 2 above)
Comment (tsehelnyk) [reply fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/service/TransactionService.java:36]: Deleted unnecessary vars, changed fields to null.

File: firefly-exporter/src/main/java/de/adorsys/opba/fireflyexporter/service/FireFlyTransactionExporter.java (hunk 3 above)
Comment (tsehelnyk) [reply firefly-exporter/src/main/java/de/adorsys/opba/fireflyexporter/service/FireFlyTransactionExporter.java:54]: Deleted unnecessary vars, changed fields to null.

Comment (tsehelnyk) [reply firefly-exporter/src/main/java/de/adorsys/opba/fireflyexporter/service/FireFlyTransactionExporter.java:54]: fixed

File: opba-protocols/xs2a-protocol/src/main/java/de/adorsys/opba/protocol/xs2a/service/xs2a/dto/Xs2aTransactionParameters.java (hunk 4 above)
Comment (tsehelnyk) [reply opba-protocols/xs2a-protocol/src/main/java/de/adorsys/opba/protocol/xs2a/service/xs2a/dto/Xs2aTransactionParameters.java:52]: fixed

File: opba-protocols/xs2a-protocol/src/main/java/de/adorsys/opba/protocol/xs2a/service/xs2a/dto/Xs2aTransactionParameters.java (hunk 5 above)
Comment (tsehelnyk) [reply opba-protocols/xs2a-protocol/src/main/java/de/adorsys/opba/protocol/xs2a/service/xs2a/dto/Xs2aTransactionParameters.java:58]: fixed

File: opba-banking-rest-api-ymls/src/main/resources/static/tpp_banking_api_ais.yml [hunk 11]
```diff
@@ -551,6 +590,41 @@ components:
...
+          description: Current page number
+          type: integer
+          format: int32
+          minimum: 1
+          example: 1
+        perPage:
+          description: Current page size (number of entries in this page)
+          type: integer
+          format: int32
+          minimum: 1
```
[the remark this answers is not in the document]
Comment (tsehelnyk) [reply opba-banking-rest-api-ymls/src/main/resources/static/tpp_banking_api_ais.yml:607]: api fixed to receive min page number `0`, 
removed **_default page number_** and **_records per page_** values for getTransactions request

**Review by valb3r — APPROVED** [review]

**Pull request merged**